source/count.js

```diff
@@ -0,0 +1,26 @@
+import reduce from './reduce.js';
```

Comment:
I think it would be better to use the internal/_reduce function here. That way we aren't dealing with all the wrapper stuff.

Reply:
It seems that it's not a convention to choose internal functions over public ones.

Since it's not a convention, I don't think it's generally a good idea to pick internal functions when there're public available.

The problem with using internal functions is that they don't guarantee an API whereas public ones do (because they are actually consumed by users). In terms of optimisation, the only overhead is the currying. I don't believe it's worth it :)

Reply:
There is no danger in using internal functions because we have extensive testing of the API that will prevent regressions even if internal functions change. While it has been standard practice for public functions to rely on other public functions, it's something we'd like to move away from instead of continuing the practice.

Reply:
Agreed.  We added the internal helpers mostly as a performance boost, and I would stick with them when they're available (and sometimes add them when they're not.)

Reply:
DONE

used internal `_reduce`. Tbf, the overhead it removes is just currying. It's hard to imagine a scenario where that would make any impact :)